lib/ui/compositing.dart

```diff
@@ -61,6 +61,18 @@ abstract class _EngineLayerWrapper implements EngineLayer {
 
   EngineLayer _nativeLayer;
 
+  bool _debugDisposed = false;
+
+  @override
+  void dispose() {
+    assert(!_debugDisposed);
```

Comment:
Would it make sense to set `_nativeLayer` to null after disposing instead of maintaining the `_debugDisposed` flag?

Reply:
I guess we could make it nullable and avoid the flag. I can change that.

Reply:
Done - I was worried this would add a ton of null checks, but we basically treat it as nullable in most places anyway.